src/Sylius/Component/Core/OrderProcessing/OrderAdjustmentsClearer.php

```diff
@@ -21,8 +21,10 @@ final class OrderAdjustmentsClearer implements OrderProcessorInterface
 {
     private array $adjustmentsToRemove;
 
-    public function __construct(array $adjustmentsToRemove = [])
-    {
+    public function __construct(
+        array $adjustmentsToRemove = [],
+        private string $validState = OrderInterface::STATE_CART
```

Comment:
Maybe there should be an array as there is a possibility that more than one state will be the valid one? The same applies to other processors, wdyt?